test/metabase/api/setup_test.clj

```diff
@@ -409,56 +407,136 @@
-                             (update :title str)))}))))
+    (with-redefs [api.setup/state-for-checklist (constantly default-checklist-state)]
+      (is (partial= [{:name  "Get connected"
+                      :tasks [{:title        "Add a database"
+                               :completed    true
+                               :triggered    true
+                               :is_next_step false}
+                              {:title        "Set up email"
+                               :completed    true
+                               :triggered    true
+                               :is_next_step false}
+                              {:title        "Set Slack credentials"
+                               :completed    false
+                               :triggered    true
+                               :is_next_step true}
+                              {:title        "Invite team members"
+                               :completed    true
+                               :triggered    true
+                               :is_next_step false}]}
+                     {:name  "Productionize"
+                      :tasks [{:title "Switch to a production-ready app database"}]}
+                     {:name  "Curate your data"
+                      :tasks [{:title        "Hide irrelevant tables"
+                               :completed    false
+                               :triggered    false
+                               :is_next_step false}
+                              {:title        "Organize questions"
+                               :completed    true
+                               :triggered    false
+                               :is_next_step false}
+                              {:title        "Create metrics"
+                               :completed    true
+                               :triggered    false
+                               :is_next_step false}
+                              {:title        "Create segments"
+                               :completed    true
+                               :triggered    false
+                               :is_next_step false}]}]
+                    (for [{group-name :name, tasks :tasks} (mt/user-http-request :crowberto :get 200 "setup/admin_checklist")]
+                      {:name  (str group-name)
+                       :tasks (for [task tasks]
+                                (-> (select-keys task [:title :completed :triggered :is_next_step])
+                                    (update :title str)))}))))
+    (testing "info about switching to postgres or mysql"
+      (testing "is included when h2 and not hosted"
+        (with-redefs [api.setup/state-for-checklist (constantly default-checklist-state)]
+          (let [checklist (mt/user-http-request :crowberto :get 200 "setup/admin_checklist")]
+            (is (= ["Get connected" "Productionize" "Curate your data"]
```

Comment:
not sure checking the group name is helpful in this case, I think we want to check the tasks title.

Reply:
the structure is
```clojure
[{:name "Get connected"
  :tasks [...]}
 {:name "Productionize"
  :tasks [...]}
 {:name "Curate your data"
  :tasks [...]}]
```

So the title of the group names goes hand in hand with the tasks they bring along. The "Productionize" group is omitted when the instance is hosted and that's all I'm testing here.

If that doesn't sway you I can get a set of all task titles and check that the migrate off of h2 task is not present. Interested to see if you think that's necessary as this feels fine to me.